lib/libzfs_core/libzfs_core.c

```diff
@@ -927,15 +946,15 @@ int lzc_receive_one(const char *snapname, nvlist_t *props,
  * this nvlist
  */
 int lzc_receive_with_cmdprops(const char *snapname, nvlist_t *props,
-    nvlist_t *cmdprops, const char *origin, boolean_t force,
-    boolean_t resumable, boolean_t raw, int input_fd,
+    nvlist_t *cmdprops, uint8_t *wkeydata, uint_t wkeylen, const char *origin,
+    boolean_t force, boolean_t resumable, boolean_t raw, int input_fd,
```

Comment:
Although we haven't officially frozen the libzfs_core interfaces, I'd like to recommend that we keep existing interfaces the same where possible.  In this case it seems straightforward to add a new variant of `lzc_receive_*` that takes the additional arguments.

Reply:
I agree that in general we should consider the libzfs_core interface frozen, and I don't want to make a habit of changing it.  But in this one specific case I think it would be reasonable to make an exception for a few reasons.

The `lzc_receive_with_cmdprops()` function is currently Linux specific first appearing in the  0.7.x releases.  No other OpenZFS platforms would be impacted.  Additionally the native encryption changes already modified this interface post 0.7.x so that ship has already sailed to some degree.  Since no issues were reported when that change was made there probably aren't many, if any, non-libzfs consumers.  I'd rather get it as complete as possible now so there's only a single version to maintain long term.

Reply:
the in-tree consumer `libzfs_core.py` and its tester `test_libzfs_core.py` needs to be changed to be compatible